File: leetcode/242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        sDict: dict[str, int] = {}
        tDict: dict[str, int] = {}

        for i in range(len(s)):
            sDict[s[i]] = sDict.get(s[i], 0) + 1
            tDict[t[i]] = tDict.get(t[i], 0) + 1
        
        for char, count in sDict.items():
            if tDict.get(char) != count:
                return False
        return True

File: leetcode/242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        int count[26] = {0};
        for(char i : s) {
            count[i - 'a']++;
        }
        for (char i : t) {
            count[i - 'a']--;
        }
        for (int i : count) {
            if(i!=0) {
                return false;
            }
        }
        return true;
    }
};